test/python/opflow/test_gradients.py

```diff
@@ -876,6 +876,46 @@ def test_gradient_wrapper(self, backend_type):
             self.assertTrue(np.allclose(result[0], correct_values[i][0], atol=0.1))
             self.assertTrue(np.allclose(result[1], correct_values[i][1], atol=0.1))
 
+    @idata(['statevector_simulator', 'qasm_simulator'])
+    def test_gradient_wrapper2(self, backend_type):
+        """Test the gradient wrapper for gradients checking that statevector and qasm gives the
+           same results
+
+        dp0/da = cos(a)sin(b) / 2
+        dp1/da = - cos(a)sin(b) / 2
+        dp0/db = sin(a)cos(b) / 2
+        dp1/db = - sin(a)cos(b) / 2
+        """
+        method = 'lin_comb'
+        a = Parameter('a')
+        b = Parameter('b')
+        params = [a, b]
+
+        qc = QuantumCircuit(2)
+        qc.h(1)
+        qc.h(0)
+        qc.sdg(1)
+        qc.cz(0, 1)
+        qc.ry(params[0], 0)
+        qc.rz(params[1], 0)
+        qc.h(1)
+
+        obs = (Z ^ X) - (Y ^ Y)
+        op = ~StateFn(obs) @ CircuitStateFn(primitive=qc, coeff=1.)
+
+        shots = 100000
+
+        values = [[0, np.pi / 2], [np.pi / 4, np.pi / 4], [np.pi / 3, np.pi / 9]]
+        correct_values = [[-2., 1.12e-16], [-1., -2.41421356], [-0.34202014, -3.5069806]]
```

Comment:
might be a bit easier to read
```suggestion
        correct_values = [[-2., 0], [-1., -2.41421356], [-0.34202014, -3.5069806]]
```